workers/data_refinery_workers/downloaders/array_express.py

```diff
@@ -108,20 +115,57 @@ def download_array_express(job_id: int) -> None:
         # filename is a problem. However, I need to know more about the naming convention.
         try:
-            original_file = OriginalFile.objects.filter(source_filename=og_file['filename']).order_by('created_at')[0]
-            original_file.is_downloaded=True
-            original_file.is_archive=False
+            original_file = OriginalFile.objects.filter(
+                source_filename=og_file['filename']).order_by('created_at')[0]
+            original_file.is_downloaded = True
+            original_file.is_archive = False
             original_file.absolute_file_path = og_file['absolute_path']
             original_file.filename = og_file['absolute_path'].split('/')[-1]
             original_file.calculate_size()
-            original_file.calculate_sha1()
             original_file.save()
+            original_file.calculate_sha1()
             og_files.append(original_file)
         except Exception:
             # TODO - is this worth failing a job for?
-            logger.warn("Found a file we didn't have an OriginalFile for! Why did this happen?: " + og_file['filename'],
+            logger.warn(
+                "Found a file we didn't have an OriginalFile for! Why did this happen?: " +
+                    og_file['filename'],
+                        downloader_job=job_id)
+            continue
+
+        sample_objects = Sample.objects.filter(originalfile=original_file).order_by('created_at')
+        if sample_objects.count() > 1:
+            logger.warn("Found an Array Express OriginalFile with more than one sample: %s",
+                        filename,
                         downloader_job=job_id)
-    success=True
+
+        # The .CEL file is the ultimate source of truth about the sample's platform.
+        sample_object = sample_objects[0]
+        if sample_object.has_raw:
+            try:
+                external_platform = microarray.get_platform_from_CEL(
+                    original_file.absolute_file_path)
+
+                platform_accession_code = "UNSUPPORTED"
+                for platform in get_supported_microarray_platforms():
+                    if platform["external_accession"] == external_accession:
+                        platform_accession_code = platform["platform_accession"]
+
+                if platform_accession_code == "UNSUPPORTED":
+                    logger.error("Found a raw file with an unsupported platform!",
+                                 file_name: filename,
+                                 sample=sample_id,
+                                 downloader_job=job_id)
+                    job.failure_reason = "Found a raw file with an unsupported platform."
+                    success = False
+
+                sample_object.platform_accession_code = platform_accession_code
+                sample_object.platform_name = get_readable_platform_names()[platform_accession_code]
+                sample_object.save()
+            except Exception:
```

Comment:
Is warning enough here? Does this indicate that we should throw this file out?

Reply:
TBH I'm not sure. It seems to make sense to me that if the header is corrupted then the rest of the file would be corrupted too. However, I suppose the header could also just be misformatted or otherwise mangled enough that something attempting to read the header couldn't, but the rest of the file could still be usable?

I figured that if the file wasn't usable then the processor job would fail and if we saw a bunch failing and never had one go through successfully, then we could change this. However I don't feel strongly about this either way so if you think we should throw this out I'm game to make the change.